### PR TITLE
Block Docker container IMDS access

### LINKED ISSUE #288: Block Docker IMDS access on Batch node

### PR DESCRIPTION
**Pull request by MattMcL4475**
Resolves #288

### CODE REVIEW

**Comment by BMurri** [comment]
Needs an open issue

File: src/Tes.Runner/Docker/NetworkUtility.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            await RunIptablesCommandAsync(addCommand, ipAddress, "blocking", callerMemberName);
+        }
+
+        private async Task RemoveBlockRuleAsync(string ipAddress, string callerMemberName)
+        {
+            string removeCommand = $"-D DOCKER-USER -i eth0 -o eth0 -m conntrack --ctorigdstaddr {ipAddress} -j DROP";
+            await RunIptablesCommandAsync(removeCommand, ipAddress, "unblocking", callerMemberName);
+        }
+
+        private async Task RunIptablesCommandAsync(string arguments, string ipAddress, string action, string callerMemberName)
```
Comment (giventocode) [inline src/Tes.Runner/Docker/NetworkUtility.cs:89]: You could return the output on this method, and use it to execute the command that checks if the IP is blocked.

File: src/Tes.Runner/Docker/NetworkUtility.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                    RedirectStandardError = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                }
+            };
+
+            process.Start();
+            await process.WaitForExitAsync();
+
+            string output = process.StandardOutput.ReadToEnd();
```
Comment (giventocode) [inline src/Tes.Runner/Docker/NetworkUtility.cs:107]: Use the async version of the method?

**Review by giventocode — APPROVED** [review]
LGTM. Minor suggestions.

**Pull request merged**